### PR TITLE
refactor devtools_server to minimize transitive deps

### PR DESCRIPTION
**Pull request by devoncarew**
- refactor devtools_server to minimize transitive deps
- fix https://github.com/flutter/devtools/issues/2958

This separates code out of package:devtools_server (and into package:devtools_shared) to allow code in the dart sdk (DDS) to depend on package:devtools_shared.

There are a lot of use cases that are supported by package:devtools, package:devtools_server, and package:devtools_shared, so it's hard to make changes to this area.

I'm not committed to this specific refactoring of the packages. We could also choose to instead move some code into package:devtools, and have package:devtools_server retain just the code that DDS needs to depend on.

### CODE REVIEW

**Comment by devoncarew** [comment]
cc @bkonyi

**Comment by bkonyi** [comment]
This looks perfect! Pulling the common server functionality into `devtools_shared` makes sense to me.

File: packages/devtools_server/lib/src/memory_profile.dart [hunk 1]
```diff
@@ -9,7 +9,7 @@ import 'package:devtools_shared/devtools_shared.dart';
 import 'package:intl/intl.dart' as intl;
 import 'package:vm_service/vm_service.dart';
 
-import 'service_registrations.dart' as registrations;
```
Comment (kenzieschmoll) [inline packages/devtools_server/lib/src/memory_profile.dart:12]: I don't think this file rename makes sense, as not all of the service code in service_registrations.dart are pertaining to memory features

File: packages/devtools_shared/pubspec.yaml [hunk 2]
```diff
@@ -11,4 +11,7 @@ environment:
   sdk: '>=2.12.0 <3.0.0'
 
 dependencies:
+  path: ^1.8.0
```
Comment (kenzieschmoll) [inline packages/devtools_shared/pubspec.yaml:14]: should we be able to remove these deps from the devtools_server pubspec now?

**Comment by kenzieschmoll** [comment]
Note that this change will require some mirrored modifications to the google3 entry point, but I don't see any red flags.

File: packages/devtools_server/lib/src/memory_profile.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools_server/lib/src/memory_profile.dart:12]: OK, I reverted the rename.

File: packages/devtools_shared/pubspec.yaml (hunk 2 above)
Comment (devoncarew) [reply packages/devtools_shared/pubspec.yaml:14]: I'll verify which deps are actually necessary now.

Comment (devoncarew) [reply packages/devtools_shared/pubspec.yaml:14]: I was able to remove the dep on package:usage.

**Pull request merged**